pkg/utils/pod/scheduling.go

```diff
@@ -58,6 +121,20 @@ func IsTerminating(pod *v1.Pod) bool {
 	return pod.DeletionTimestamp != nil
 }
 
+func IsStuckTerminating(pod *v1.Pod, clk clock.Clock) bool {
+	// The pod DeletionTimestamp will be set to the time the pod was deleted plus its
+	// grace period in seconds. We give an additional minute as a buffer to allow
+	// pods to force delete off the node before we actually go and deprovision the node
```

Comment:
nit: i think we should migrate off the deprovision vocab since we are using disrupt/terminate
```suggestion
	// pods to force delete off the node before we actually go and terminate the node
```